### PR TITLE
Clean up code, Reorganize testscripts

### PR DESCRIPTION
**Pull request by jhlee-mitre**
# Summary
1. Clean up source code
2. Reorganize testsciprts. Keep only 5 example testscripts from the spec under default path (./TestScripts) Moved rest to ./spec/examples for testing / validation purposes
3. Refine messages and spaces.
4. Reorganize spec folder and files
5. This PR has no impact on execution of the engine

### CODE REVIEW

**Review by karlnaden — APPROVED** [review]
no concerns

File: config.yml [hunk 1]
```diff
@@ -1,38 +1,35 @@
-# Whether to ignore non-FHIR fixtures
+# If true, allow intake non-FHIR fixture (local only)
 nonfhir_fixture: true
 
 # Endpoint against which TestScripts will be executed
 server_url: https://server.fire.ly
 
-# The relative path to the directory containing the TestScript resources (as JSON or XML) to be executed by the engine
-testscript_path: "/spec/examples/testscript_unittest"
+# Relative path to the directory containing the TestScript resources (as JSON or XML) to be executed by the engine
```
Comment (karlnaden) [inline config.yml:7]: Currently requires a leading `/` character, which won't be expected. Recommend expecting no leading `/` for this and the testreport_path

File: config.yml [hunk 2]
```diff
@@ -1,38 +1,35 @@
...
-- PatientResourceId2 = var2changed
+- name1=value1
+- name2=value2
 
-# The relative to the directory containing the TestReports output following their partner TestScript execution
+# Relative to the directory containing the TestReports output following their partner TestScript execution
 testreport_path: "./TestReports"
 
-# Whether use FHIR logger
+# If true use FHIR logger
```
[the remark this answers is not in the document]
Comment (jhlee-mitre) [reply config.yml:26]: FHIR logger is currently used in two places:

```
  def valid_resource?(resource, *profiles, options)
    FHIR.logger = Logger.new(STDOUT) if options["verbose"]

    validate_using_operation(resource, profiles)
    validate_using_route(resource, profiles) unless reply.response[:code].start_with?('2')
    !validation_errors?
  end
```

and Ruby unit testing: `spec_helper.rb`

The message you see on console isn't about FHIR logger, not that I know. FHIR log is exported to a file. You can set below to see how it logs. Should you find log.txt in your working directory.

`FHIR.logger = Logger.new("log.txt") `

About the message you point above, I wonder how I can remove those messages whenever executing FHIR clients. It make console quite messy. FYI Jack and I wanted to have a kind console explaining what happened during execution, whereas Jay's preference was put useful information into TestReport outcomes and doesn't much care of console.

File: config.yml (hunk 1 above)
Comment (jhlee-mitre) [reply config.yml:7]: Good point. Also I don't think we need quotation in config.yml. I will clean it up and recommit.

Comment (karlnaden) [reply config.yml:7]: You probably need the quotes if your path contains spaces, otherwise not

Comment (jhlee-mitre) [reply config.yml:7]: Right

**Pull request merged**